### ISSUE TITLE
Latest python version

### ISSUE THREAD

**Issue by arecker** [netlify/build-image#606]
Hello!  The python versions available on the netlify build image could use some love.  The current [stable version is 3.9.6](https://www.python.org/downloads/), but the highest netlify goes is 3.7.  Any chance you could update it?

**Comment by sambacha** [comment]
Edit: default branch of `xenial` confused me, as `focal` is what all buildimages use, and if you check the supported software, python 3.8 is listed.
https://github.com/netlify/build-image/blob/focal/included_software.md

**Comment by Stanzilla** [comment]
Any plans to add 3.10?

**Comment by kitop** [comment]
Are you able to install newer versions by using the `PYTHON_VERSION` environment variable or one of the other options? https://docs.netlify.com/configure-builds/manage-dependencies/#python

**Comment by merwok** [comment]
That envvar does not install arbitrary version, it only selects one of the available options.

**Comment by Stanzilla** [comment]
3.11 would be nice!

**Comment by ar1ja** [comment]
Python 3.10 and/or python3.11 would be nice
https://answers.netlify.com/t/python-3-10-3-11-or-newer-in-netlify-ci/78648

**Comment by kitop** [comment]
We're looking into improving version support in general, but don't have any ETA for it.

Have you tried using Homebrew to install these newer versions?

**Comment by ar1ja** [comment]
> We're looking into improving version support in general, but don't have any ETA for it.
> 
> Have you tried using Homebrew to install these newer versions?

don't ya think it's gonna waste a lot of build minutes though

**Comment by Stanzilla** [comment]
> We're looking into improving version support in general, but don't have any ETA for it.
> 
> Have you tried using Homebrew to install these newer versions?

I don't think that works well on Linux right now and also what ^ they said.

**Comment by ar1ja** [comment]
> > We're looking into improving version support in general, but don't have any ETA for it.
> > Have you tried using Homebrew to install these newer versions?
> 
> I don't think that works well on Linux right now and also what ^ they said.

Yeah, homebrew is meant of mac and it works kinda garbo on Linux, especially a distro like Ubuntu, I think the devs should just supply the latest image(s) (potentially even automating this updating process ?), or even let users to set an arbitrary python version (maybe it could be like:

```py
if build.SET_PYTHON_VERSION not in IMAGE_DB:
    logging.log(f"{SET_PYTHON_VERSION!r} not in saved images database, please wait for us to set up the image on our database, this will take around 20 minutes")
    build.fail()
    IMAGE_DB.save(create_image(SET_PYTHON_VERSION))
else:
    build()
```
)

**Comment by Mara-Li** [comment]
Hello.
Any news about this? I'm afraid that, if time pass, my dependencies become unusable because of this.

**Comment by jarofgreen** [comment]
3.11 would be great, just for this:

"Python 3.11 is between 10-60% faster than Python 3.10. On average, we measured a 1.25x speedup on the standard benchmark suite." https://docs.python.org/3/whatsnew/3.11.html

**Comment by orsinium** [comment]
Python 3.10+ is a must for my use case:

I use netlify to host Sphinx-powered documentation for my OSS projects. The issue is that I use the new style of type annotations (generics for built-in types and union operator) that require Python 3.10+. The project itself works on Python 3.8+ but Sphinx should be able to parse type annotations to properly render links to internal types in them. And it cannot be done on Python 3.8 or 3.9. I could rewrite all type annotations to the old style just for netlify but that's not cool.

**Comment by ar1ja** [comment]
> Python 3.10+ is a must for my use case:
> 
> I use netlify to host Sphinx-powered documentation for my OSS projects. The issue is that I use the new style of type annotations (generics for built-in types and union operator) that require Python 3.10+. The project itself works on Python 3.8+ but Sphinx should be able to parse type annotations to properly render links to internal types in them. And it cannot be done on Python 3.8 or 3.9. I could rewrite all type annotations to the old style just for netlify but that's not cool.

3.11 and up would be nice bc of the performance improvements
